scvi/model/base/_pyromixin.py

```diff
@@ -0,0 +1,472 @@
+
+        """
+
+        samples = dict()
+
+        gpus, device = parse_use_gpu_arg(use_gpu)
+
+        if batch_size is None:
+            batch_size = self.adata.n_obs
+        train_dl = AnnDataLoader(self.adata, shuffle=False, batch_size=batch_size)
+        # sample local parameters
+        i = 0
+        for tensor_dict in track(
+            train_dl,
+            style="tqdm",
+            description="Sampling local variables, batch: ",
+        ):
+            args, kwargs = self.module._get_fn_args_from_batch(tensor_dict)
+            args = [a.to(device) for a in args]
+            kwargs = {k: v.to(device) for k, v in kwargs.items()}
+            self.to_device(device)
+
+            if i == 0:
+                obs_plate_sites = self._get_obs_plate_sites(args, kwargs)
+                if len(obs_plate_sites) == 0:
+                    # if no local variables - don't sample
+                    break
+                obs_plate_dim = list(obs_plate_sites.values())[0]
+
+                sample_kwargs_obs_plate = sample_kwargs.copy()
+                sample_kwargs_obs_plate[
+                    "return_sites"
+                ] = self._get_obs_plate_return_sites(
+                    sample_kwargs, list(obs_plate_sites.keys())
+                )
+                sample_kwargs_obs_plate["show_progress"] = False
+
+                samples = self._get_posterior_samples(
+                    args, kwargs, **sample_kwargs_obs_plate
+                )
+            else:
+                samples_ = self._get_posterior_samples(
+                    args, kwargs, **sample_kwargs_obs_plate
+                )
+
+                samples = {
+                    k: np.array(
+                        [
+                            np.concatenate(
+                                [samples[k][j], samples_[k][j]],
+                                axis=obs_plate_dim,
+                            )
+                            for j in range(
+                                len(samples[k])
+                            )  # for each sample (in 0 dimension
+                        ]
+                    )
+                    for k in samples.keys()  # for each variable
+                }
+            i += 1
+
+        # sample global parameters
+        global_samples = self._get_posterior_samples(args, kwargs, **sample_kwargs)
+        global_samples = {
+            k: v
+            for k, v in global_samples.items()
+            if k not in list(obs_plate_sites.keys())
+        }
+
+        for k in global_samples.keys():
+            samples[k] = global_samples[k]
+
+        self.module.to(device)
+
+        return samples
+
+    def sample_posterior(
+        self,
+        num_samples: int = 1000,
+        return_sites: Optional[list] = None,
+        use_gpu: bool = None,
+        batch_size: int = 128,
+        sample_kwargs=None,
+        return_samples: bool = False,
+        summary_fun: Optional[dict] = None,
```

Comment:
```suggestion
        summary_fun: Optional[Dict[str, Callable]] = None,
```
you might need to import Dict and Callable from typing